app/views/shared/_sp_alert.html.slim

```diff
@@ -1,10 +1,13 @@
-- if decorated_session.sp_alert?
-  - sp_alert_name = decorated_session.sp_alert_name
+- sp_alert_name = decorated_session.sp_alert_name
+- has_sign_up_text = I18n.exists?("service_providers.#{sp_alert_name}.create_account_page.body")
+- if decorated_session.sp_alert? && !(current_page?(sign_up_email_path) && !has_sign_up_text)
```

Comment:
Why not create an empty `create_account_page.body` entry for those SPs that don't need one? This will eliminate these conditionals which are hard to read IMO. It is generally considered a bad practice to have conditionals in views. We prefer the "tell don't ask" principle, such as in our heavy use of presenters, which could be another option here.

Reply:
That would imply creating a blank entry for every sp in the yml file as well as every future sp and translation when on average it will not be supplied.  I agree that it's bad style to put these conditionals in the view but the conditionals were there before I got here. The correct way to do this would be to pull out all the sp specific code from the views and session decorator (we already have the issuer which is unique) and drive it with some data configuration that feeds the decorated session which feeds the view. You are proposing to use the locale files as a mechanism to configure what gets shown.  I can do it but I'd prefer 2 lines of code over dozens of unused lines in the locale files.

Reply:
Fair enough.

Reply:
I'm not totally against it.  At least it makes it more data bound.  I think the specs would need to be updated because they are testing for an alert box.  I'm not sure if the alert box is invisible or not when there is no text.